test/heap-types.wast

```diff
@@ -0,0 +1,24 @@
+;; Test that we can roundtrip struct and array types
+(module
+  (type $struct.A (struct
+    i32
```

Comment:
I'm surprised that `field` is optional. Does the explainer have an example like this?

Reply:
The [very first example](https://github.com/WebAssembly/gc/blob/master/proposals/gc/Overview.md#type-grammar) does:
```wat
(type $tup (struct i64 i64 i32))
```
I agree in general that the large amount of sugar in the wasm text format can be kind of annoying when writing a parser... but I guess it helps when writing code which is more frequent.